### PR TITLE
Fixing valueToBlock method

### LINKED ISSUE #8: Tests are red because a byte symbol DNU #cachePragmas message
A lot of tests are red because the message #cachePragmas is sent to a byte symbol.

This message is sent during a step, when the interpreter visits a message send,`DASTContext class` selects some associations in the `BlockClosure methodDict` and then collects the selectors of the collected methods, as if the result of the `select:`expression was a collection of methods instead of a method dictionary:

image

As a result, the `collect:` expression returns a method dictionary with selectors (byte symbols) as values instead of compiled methods. That explains the DNU:

image

### PR DESCRIPTION
**Pull request by adri09070**
Fixes #8

`DASTContext class>>#valueToBlock` used a `collect:` expression on a MethodDictionary, that returns a MethodDictionary which  should contain compiled methods as values but which contain selectors instead. This was producing a DNU.

As its senders of this method were only using method selectors, I fixed it so that it returns only an array  of selectors instead of a method dictionary.

Furthermore, this method should return the selector that use primitives that enter blocks (i.e: `#value` and its variants). However, the IDs of these primitives have changed, which were used by this method. So, instead of updating the IDs of these primitives in this method, I refactored the method so that it checks the selector of the methods that use this primitive instead, as these selectors are less likely to change in the future

### CODE REVIEW

File: DebuggableASTInterpreter/DASTContext.class.st [hunk 1]
```diff
@@ -68,8 +68,17 @@ DASTContext class >> newWithSender: aDASTContext receiver: aReceiver messageNode
...
+	^ valueToBlockMessages ifNil: [ 
+		  valueToBlockMessages := (BlockClosure methodDict select: [ :e | "Is it really better than `(205 to: 209) includes: e primitive` ?"
+			                           #( #value #value: #value:value:
+			                              #value:value:value:
+			                              #value:value:value:value:
+			                              #value #value:value: #value:value:
+			                              #valueNoContextSwitch
+			                              valueNoContextSwitch:
+			                              #valueWithArguments ) includes:
+				                           e selector ]) keys ]
```
Comment (guillep) [inline DebuggableASTInterpreter/DASTContext.class.st:81]: Why do you need to select, then keys? why not just returning the list of selectors? :)

**Review by guillep — CHANGES_REQUESTED** [review]
comment

File: DebuggableASTInterpreter/DASTContext.class.st (hunk 1 above)
Comment (adri09070) [reply DebuggableASTInterpreter/DASTContext.class.st:81]: Indeed, I'm dumb :) . I was to focused on the original expression

**Comment by adri09070** [comment]
This is done

**Review by guillep — APPROVED** [review]
Good!

**Comment by guillep** [comment]
Thanks!

**Pull request merged**